core/js_errors.rs

```diff
@@ -41,77 +43,8 @@ pub struct JSError {
   pub frames: Vec<StackFrame>,
 }
 
-impl std::error::Error for JSError {
-  fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
-    None
-  }
-}
-
-impl fmt::Display for StackFrame {
-  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-    // Note when we print to string, we change from 0-indexed to 1-indexed.
-    let function_name = self.function_name.clone();
-    let script_line_column =
-      format_script_line_column(&self.script_name, self.line, self.column);
-
-    if !self.function_name.is_empty() {
-      write!(f, "    at {} ({})", function_name, script_line_column)
-    } else if self.is_eval {
-      write!(f, "    at eval ({})", script_line_column)
-    } else {
-      write!(f, "    at {}", script_line_column)
-    }
-  }
-}
-
-fn format_script_line_column(
-  script_name: &str,
-  line: i64,
-  column: i64,
-) -> String {
-  // TODO match this style with how typescript displays errors.
-  let line = (1 + line).to_string();
-  let column = (1 + column).to_string();
-  let script_name = script_name.to_string();
-  format!("{}:{}:{}", script_name, line, column)
-}
-
-impl fmt::Display for JSError {
-  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-    if self.script_resource_name.is_some() {
-      let script_resource_name = self.script_resource_name.as_ref().unwrap();
-      if self.line_number.is_some() && self.start_column.is_some() {
-        assert!(self.line_number.is_some());
-        assert!(self.start_column.is_some());
-        let script_line_column = format_script_line_column(
-          script_resource_name,
-          self.line_number.unwrap() - 1,
-          self.start_column.unwrap() - 1,
-        );
-        write!(f, "{}", script_line_column)?;
-      }
-      if self.source_line.is_some() {
-        write!(f, "\n{}\n", self.source_line.as_ref().unwrap())?;
-        let mut s = String::new();
-        for i in 0..self.end_column.unwrap() {
-          if i >= self.start_column.unwrap() {
-            s.push('^');
-          } else {
-            s.push(' ');
-          }
-        }
-        writeln!(f, "{}", s)?;
-      }
-    }
-
-    write!(f, "{}", self.message.clone())?;
-
-    for frame in &self.frames {
-      write!(f, "\n{}", &frame.to_string())?;
-    }
-    Ok(())
-  }
-}
+#[derive(Debug, PartialEq, Clone)]
+pub struct CoreJSError(V8Exception);
```

Comment:
What's the purpose of wrapping V8Exception?

Reply:
Discussed out-of-band. V8Exception is not an Error type, just POD. CoreJSError implements Error.